web/src/main/java/org/cbioportal/web/StudyViewController.java

```diff
@@ -303,6 +305,31 @@
         return new ResponseEntity<>(result, HttpStatus.OK);
     }
 
+    @PreAuthorize("hasPermission(#involvedCancerStudies, 'Collection<CancerStudyId>', 'read')")
```

Comment:
You also will need to add to [InvolvedCancerStudyExtractorInterceptor](https://github.com/cBioPortal/cbioportal/blob/f1bde3980dbe3945e4d0a835c0dffefb9ffeba52/web/src/main/java/org/cbioportal/web/util/InvolvedCancerStudyExtractorInterceptor.java) (probably ok to match /cna-genes/fetch).

Reply:
will take a look at. thanks!